odoo/addons/base/models/ir_http.py

```diff
@@ -487,3 +490,34 @@ def _response_by_status(self, status, headers, content):
             return werkzeug.utils.redirect(content, code=301)
         elif status != 200:
             return request.not_found()
+
+    @classmethod
+    def _get_session_store(cls):
+        path = odoo.tools.config.session_dir
+        _logger.debug('HTTP sessions stored in: %s', path)
+        return sessions.FilesystemSessionStore(
+            path, session_class=OpenERPSession, renew_missing=True)
```

Comment:
Should call into `odoo.http.root.session_store` instead of duplicating it.

In case of fear that the WSGI application would be replaced (e.g. a middleware would be put around it, which would be the case if sentry is used for instance) `Root.session_store` should probably be converted to a `lazy_classproperty`.

Reply:
I considered that, but having it in ir_http means we can override it in modules. It makes it easier for a custom module to define its own session_store. 

If I had to remove one or the other, I'd definitely drop the Root.session_store and replace it to pull the value from ir_http. The thing I'm unsure is how it would work with HttpWorkers, having it in ir_http makes it depends on the modules being loaded and ir.config_parameter makes it depend on the database being used. From a cron perspective, the necessary data is already loaded. But from the HttpRequest perspective, we're trying to load the session that may contain information on which database to use so ir.config_parameter can't be used since the database choice can't be made already as we don't know which session or how to load it.

Having it in Root.session_store is at least consistent but a bit limiting. It could be fixed later thought.